src/tracer/tracer.ts

```diff
@@ -49,70 +51,86 @@ export const LEAK_MESSAGE =
-      });
-      ExecutionTags.autoTagEvent(event);
-    } catch (err) {
-      logger.warn('Failed to start tracer', err);
-    }
+      if (!context || !isAwsContext(context)) {
+        logger.warnClient(
+          'missing context parameter - learn more at https://docs.lumigo.io/docs/nodejs'
+        );
+        const { err, data, type } = await promisifyUserHandler(userHandler, event, context);
+        return performPromisifyType(err, data, type, callback);
+      }
 
-    if (!context || !isAwsContext(context)) {
-      logger.warnClient(
-        'missing context parameter - learn more at https://docs.lumigo.io/docs/nodejs'
-      );
-      const { err, data, type } = await promisifyUserHandler(userHandler, event, context);
-      return performPromisifyType(err, data, type, callback);
-    }
+      if (context.__wrappedByLumigo) {
+        const { err, data, type } = await promisifyUserHandler(userHandler, event, context);
+        return performPromisifyType(err, data, type, callback);
+      }
+      context.__wrappedByLumigo = true;
 
-    if (context.__wrappedByLumigo) {
-      const { err, data, type } = await promisifyUserHandler(userHandler, event, context);
-      return performPromisifyType(err, data, type, callback);
-    }
-    context.__wrappedByLumigo = true;
+      const functionSpan = getFunctionSpan(event, context);
 
-    const functionSpan = getFunctionSpan(event, context);
+      await hookUnhandledRejection(functionSpan);
 
-    await hookUnhandledRejection(functionSpan);
+      const pStartTrace = startTrace(functionSpan);
+      const pUserHandler = promisifyUserHandler(userHandler, event, context);
 
-    const pStartTrace = startTrace(functionSpan);
-    const pUserHandler = promisifyUserHandler(userHandler, event, context);
+      let [, handlerReturnValue] = await Promise.all([pStartTrace, pUserHandler]);
 
-    let [, handlerReturnValue] = await Promise.all([pStartTrace, pUserHandler]);
+      handlerReturnValue = normalizeLambdaError(handlerReturnValue);
 
-    handlerReturnValue = normalizeLambdaError(handlerReturnValue);
+      if (isStepFunction()) {
+        handlerReturnValue = performStepFunctionLogic(handlerReturnValue);
+      }
 
-    if (isStepFunction()) {
-      handlerReturnValue = performStepFunctionLogic(handlerReturnValue);
-    }
+      const cleanedHandlerReturnValue = removeLumigoFromStacktrace(handlerReturnValue);
 
-    const cleanedHandlerReturnValue = removeLumigoFromStacktrace(handlerReturnValue);
+      await endTrace(functionSpan, cleanedHandlerReturnValue);
+      const { err, data, type } = cleanedHandlerReturnValue;
 
-    await endTrace(functionSpan, cleanedHandlerReturnValue);
-    const { err, data, type } = cleanedHandlerReturnValue;
+      return performPromisifyType(err, data, type, callback);
+    };
 
-    return performPromisifyType(err, data, type, callback);
+    if (isResponseStreamFunction) {
+      decoratedUserHandler[HANDLER_STREAMING] = STREAM_RESPONSE;
```

Comment:
Printing the code for `awslambda.streamifyResponse` shows that it actually sets another value:
```js
(handler, options) => {
            handler[HANDLER_STREAMING] = STREAM_RESPONSE;
            if (typeof options?.highWaterMark === "number") {
              handler[HANDLER_HIGHWATERMARK] = parseInt(options.highWaterMark);
            }
            return handler;
          }
```
I'm not sure exactly what does `HANDLER_HIGHWATERMARK` do, but I think we better set it as well - which brings me back to the point where we probably better do something like:

```js
if (isResponseStreamFunction) {
   return awslambda.streamifyResponse(decoratedUserHandler)
}
```
I would also check for the existence of `awslambda.streamifyResponse` even though `isResponseStreamFunction` guards us in those cases, since the implementation of `awslambda.streamifyResponse` might change in the future and us not guaranteed to rely on the existence of `HANDLER_STREAMING`. So maybe something like:
```js
if (isResponseStreamFunction) {
   const isStreamWrapperAvailable = typeof awslambda !== "undefined" && typeof awslambda.streamifyResponse == "function";
   return isStreamWrapperAvailable ? awslambda.streamifyResponse(decoratedUserHandler) : decoratedUserHandler;
}
```